### PR TITLE
Only get columns if detailViewAutoUpdate is true and device is online

### PR DESCRIPTION
**Pull request by hcarneiro**
This will fix an issue that would prevent the list from rendering when the device is offline.

### CODE REVIEW

File: js/layout-javascript/agenda-code.js [hunk 1]
```diff
@@ -706,10 +706,16 @@ DynamicList.prototype.initialize = function() {
       return;
     })
     .then(function() {
-      return Fliplet.DataSources.getById(_this.data.dataSourceId);
+      if (_this.data.detailViewAutoUpdate && Fliplet.Navigator.isOnline()) {
+        return Fliplet.DataSources.getById(_this.data.dataSourceId);
+      }
+      return
```
Comment (squallstar) [inline js/layout-javascript/agenda-code.js]: this seems unnecessary?

File: js/layout-javascript/agenda-code.js [hunk 2]
```diff
@@ -706,10 +706,16 @@ DynamicList.prototype.initialize = function() {
       return;
     })
     .then(function() {
-      return Fliplet.DataSources.getById(_this.data.dataSourceId);
+      if (_this.data.detailViewAutoUpdate && Fliplet.Navigator.isOnline()) {
+        return Fliplet.DataSources.getById(_this.data.dataSourceId);
```
Comment (squallstar) [inline js/layout-javascript/agenda-code.js]: I would suggest a `.catch(function () { return Promise.resolve(); });`

Comment (hcarneiro) [reply js/layout-javascript/agenda-code.js]: But if the error comes from somewhere else it will resolve too.

Comment (angelnikolov) [reply js/layout-javascript/agenda-code.js]: I thin he meant adding that to the `getById` call. Right?

Comment (squallstar) [reply js/layout-javascript/agenda-code.js]: @hcarneiro I'm suggesting catching this promise, not others.

```js
return Fliplet.DataSources.getById(_this.data.dataSourceId).catch(function () {
  return Promise.resolve(); // no problem when this fails
});
```

**Pull request merged**